File: DSA/ShatakshiMishra_590015008/Day9Question3.cpp
#include <iostream>
#include <unordered_map>
#include <string>
using namespace std;

char firstNonRepeatingChar(string& s) {
    unordered_map<char, int> freq;

    for (char ch : s) {
        freq[ch]++;
    }

    for (char ch : s) {
        if (freq[ch] == 1) {
            return ch;
        }
    }

    return -1; 
}

int main() {
    string input;
    getline(cin, input);

    char result = firstNonRepeatingChar(input);
    if (result != -1)
        cout << result << endl;
    else
        cout << -1 << endl;

}

File: DSA/LakshaKundra_590013237/Day_9_Question_3.py
def first_non_repeating_char(s):
    freq = {}
    for char in s:
        freq[char] = freq.get(char, 0) + 1
    for char in s:
        if freq[char] == 1:
            return char
    return -1

user_input = input("Enter a string: ")
result = first_non_repeating_char(user_input)
print("Output:", result)
